lib/services/devices/deviceRegistryMemory.js

```diff
@@ -110,16 +128,20 @@ function listDevices(service, subservice, limit, offset, callback) {
     });
 }
 
-function getDevice(id, callback) {
-    if (registeredDevices[id]) {
-        callback(null, registeredDevices[id]);
+function getDevice(id, service, subservice, callback) {
+    if (!subservice && !callback) {
+        callback = service;
+    }
+
+    if (registeredDevices[service] && registeredDevices[service][id]) {
```

Comment:
I think this may lead to unexpected results in case the callback function is passed as the `service` argument :)